python/ray/serve/_private/common.py

```diff
@@ -236,6 +236,27 @@ def __setstate__(self, d: Dict[Any, Any]) -> None:
     def set_autoscaled_num_replicas(self, autoscaled_num_replicas):
         self.autoscaled_num_replicas = autoscaled_num_replicas
 
+    def update(
+        self,
+        deployment_config: DeploymentConfig = None,
+        replica_config: ReplicaConfig = None,
+        version: str = None,
+        is_driver_deployment: bool = False,
+        route_prefix: str = None,
+    ) -> "DeploymentInfo":
+        return DeploymentInfo(
+            deployment_config=deployment_config or self.deployment_config,
+            replica_config=replica_config or self.replica_config,
+            start_time_ms=self.start_time_ms,
+            deployer_job_id=self.deployer_job_id,
+            actor_name=self.actor_name,
+            version=version or self.version,
+            end_time_ms=self.end_time_ms,
+            is_driver_deployment=is_driver_deployment or self.is_driver_deployment,
```

Comment:
there is a bug here -- if `is_driver_deployment=False` is specified in the call to `update` it won't actually get set

Reply:
Oops good catch, I've fixed it!